bundle/terranova/tnresources/adapter.go

```diff
@@ -25,12 +25,37 @@ type IResource interface {
 	// Example: func (*ResourceJob) PrepareConfig(input *resources.Job) *jobs.JobSettings
 	PrepareConfig(input any) any
 
+	// DoRefresh reads and returns remote state from the backend. The return type defines schema for remote field resolution.
+	// Example: func (r *ResourceJob) DoRefresh(ctx context.Context, id string) (*jobs.Job, error) {
+	DoRefresh(ctx context.Context, id string) (remoteState any, e error)
+
 	// DoDelete deletes the resource.
 	// Example: func (r *ResourceJob) DoDelete(ctx context.Context, id string) error {
 	DoDelete(ctx context.Context, id string) error
 
+	// [Optional] RecreateFields returns a list of fields that will cause resource recreation if changed
+	// Example: func (r *ResourceJob) RecreateFields() []string { return []string{"name", "type"} }
+	RecreateFields() []string
+
+	// [Optional] ClassifyChanges provides non-default change classification.
+	// Default is to consider any change "an update" (RecreateFields handled separately).
+	// Example: func (r *ResourceJob) ClassifyChanges(changes []structdiff.Change) deployplan.ActionType { return deployplan.ActionUpdate }
+	ClassifyChanges(changes []structdiff.Change) deployplan.ActionType
+}
+
+// This interface represents adapted interface to call a resource.
```

Comment:
Hard to parse this sentence. What's the utility of a separate interface where refresh is not supported?